core/baseline-core-v1.0-psd01.md

```diff
@@ -2309,47 +2307,249 @@ New Agreement State = Old Agreement State + Agreed upon New State Object + Works
+2. Verify that each work step instance is associated with only one workgroup through a system search that only yeilds one result for each workstep instance.
+3. Attempt to assign a work step instance to multiple workgroups.
+4. Verify that the system prevents the work step instance from being associated with more than one workgroup by creating an error warning and aborting the system process.
+5. Attempt to create a work step instance without specifying a workgroup.
+6. Verify that the system prompts the user to select a workgroup before proceeding with the creation of the work step instance.
+7. Verify that the system prevents the creation of a work step instance without associating it with a workgroup by creating an error warning and not creating the work step instance.
+8. Attempt to associate a work step instance with a non-existent workgroup.
+9. Verify that the system prompts the user to select a valid workgroup before proceeding with the association of the work step instance by creating an error warning and not proceeding with the system process..
+10. Verify that the system prevents the association of a work step instance with a non-existent workgroup by creating an error warning and aborting the system process.  
+
+If all of the above test steps are successfully completed, then the requirement is met
+
+
 #### **[R216]** 
 A workstep instance MUST inherit the security and privacy policies of its associated workgroup.
 
 *See the details of workgroups and their security and privacy policies below.*
 
+[[R216]](#r216) Testability: Policies such as an authentication requirement of a BPI Subject are expressible as functions evaluating to true or false, and, therefore, represent worksteps that can be applied to input data in a workstep instance. Furthermore, [[R215]](#r215) expresses the testable association of one or more workstep instance and a workgroup. Therefore, one or more workgroup policies can be evaluated as worktsep istances in the context of any workstep instance associated with a workgroup before that workstep instance is executed. Hence, the execution of a given workstep instance associated to a workgroup is contingent upon the output of the evaluation of one or more workgroup policies as workstep instances -- the workstep instance is executed if the individual policies together evaluate to true, or is not executed if the policies together evaluate to false either result meets the requirement. If a set of workgroup policies is not successfully associated with a workstep instance associated to the worgroup with those policies, then the workstep instance is executed independent of the evaluation result of the workgroup policies' workstep instances which does not meet the requirement.    
+
 #### **[R217]** 
 A workstep MUST have a unique identifier within a BPI.
 
+[[R217]](#r217) Testability:
+
+Preconditions:
+
+* A BPI test instance is set up and operational.
+* There are at least two worksteps set-up within the BPI.
```

Comment:
If there is only one step, just check that there is an identifier.

Reply:
Not quite. It says unique identifier (within the BPI namespace). So you need at least two to check that identifiers are different and generating two with the same identifier is not permitted.